### DISCUSSION TITLE
ft.Image not showing Local images whiles links to images on the web appear.

### DISCUSSION THREAD

**Discussion by Haadi-A** [flet-dev/flet#958] [Q&A] [answered]
Hi @ndonkoHenri and @DavMelchi. I am creating a user profile and I want to show the user's image on top of a container so I put both the picture and container in a stack.
But the problem is that the container is shown alright but the image won't show, so I thought that it was because I was using a relative path and I changed it to an absolute path and the image was still not showing so I tried to check whether it was the format because  I only see .png formats being used in the documentation, so I had to convert the image from jpg to png, but it still didn't work, then I lastly tried to use a link to an image on the web and it worked!! But when used the local image it still didn't work. I don't know why it is failing to use images on my machine. I'm also using Linux Ubuntu 22.10. 
Here is a video highlighting how this happens:

![errorvideo_in_](https://user-images.githubusercontent.com/77079200/215545183-9e84f7ea-9d68-4ad8-af9c-db82447c5461.gif)

**Comment by DavMelchi** [comment, marked as answer]
Hi @Haadi-A , share your sample code please. I'll try.

**Reply by Haadi-A** [reply]
Thanks for replying, here it is:
```
import flet as ft

def ProfilePage(page=ft.Page):
    backButton: ft.Control = ft.FilledButton(
        text="back", icon=ft.icons.ARROW_BACK_SHARP)
    page.add(
        ft.Container(content=ft.Column(
            controls=[
                ft.Row(controls=[backButton], alignment=ft.MainAxisAlignment.START,
                       ),
                ft.Stack(
                    controls=[
                    ft.Image(
                            src="/home/cisco-ramon/Documents/PROGRAMS/python/fletApps/nightMakaranta_gui/reddinton.png",
                            # src = f"https://i.pinimg.com/originals/d9/11/31/d911312b1223f5695b028a87b1a7f968.jpg",
                            width=300,
                            height=300,
                            fit=ft.ImageFit.CONTAIN,
                            repeat=ft.ImageRepeat.NO_REPEAT,
                            border_radius=ft.border_radius.all(180),
                        ),       
                    ]
                )
            ],
        ),
        ),
    )


ft.app(target=ProfilePage)
```

**Reply by DavMelchi** [reply]
```python

import flet as ft

def ProfilePage(page=ft.Page):
    backButton: ft.Control = ft.FilledButton(
        text="back", icon=ft.icons.ARROW_BACK_SHARP)
    page.add(
        ft.Container(content=ft.Column(
            controls=[
                ft.Row(controls=[backButton], alignment=ft.MainAxisAlignment.START,
                       ),
                ft.Stack(
                    controls=[
                    ft.Image(
                            # src="/home/cisco-ramon/Documents/PROGRAMS/python/fletApps/nightMakaranta_gui/reddinton.png",
                            
                            src=r"C:\Users\HP\Desktop\david.jpg",  #absolute path ok
                            # src="avatar.png", # relative path from assets folder ok
                            
                            # src = f"https://i.pinimg.com/originals/d9/11/31/d911312b1223f5695b028a87b1a7f968.jpg",
                            width=300,
                            height=300,
                            fit=ft.ImageFit.CONTAIN,
                            repeat=ft.ImageRepeat.NO_REPEAT,
                            border_radius=ft.border_radius.all(180),
                        ),       
                    ]
                )
            ],
        ),
        ),
    )


ft.app(target=ProfilePage, assets_dir="assets",) # add here 'assets_dir="assets"
```


1) Your path is relative then you should use assets folder for that.

You will learn more about `assets` folder here : https://flet.dev/docs/controls/image#src

2) I added a second example with full absolute path for you

`flet Version: 0.4.0.dev1070`

if not done yet , you can join the discord  for quick help from the community https://discord.gg/dzWXP8SHG8

**Reply by Haadi-A** [reply]
Thanks, setting the assets folder in the flet app call worked!! Thanks very much @DavMelchi.

**Reply by Haadi-A** [reply]
@DavMelchi  does the flet Version: 0.4.0.dev1070 allow you to show images using absolute paths as src ?

**Reply by DavMelchi** [reply]
Yes... As you can see in my example shared with you. Absolute path works properly.

**Reply by Haadi-A** [reply]
Yh I tried it out and absolute paths worked, but hot reload wasn't working and using relative paths to images also doesn't show them.

**Reply by ndonkoHenri** [reply]
Hot reload mode is been fixed at the moment.
Also, if you try something in hot reload mode and it doesn't work, then exit that mode and rerun your program in normal mode to see if it works.
I had this issue several times back then, and noticed quite later that I had to exit hot reload mode for everything to work as expected. 🤦🏽‍♂️

**Reply by Haadi-A** [reply]
Okay, but I'm gonna use ur sugegstion, I'm gonna create two Virt Evns.
